modules/gateway/peers.go

```diff
@@ -175,6 +175,9 @@ func (g *Gateway) Connect(addr modules.NetAddress) error {
 	if addr == g.Address() {
 		return errors.New("can't connect to our own address")
 	}
+	if build.Release != "testing" && addr.IsLoopback() {
```

Comment:
yeah, hadn't thought about using mocking to replace the constants.

'New' could include some sort of interface that defines functions for all of the constants.